### PR TITLE
Add details of ‘confirm email address’ and ‘custom retention period’ #172

### PR DESCRIPTION
**Pull request by karlchillmaid**
# Do not merge until the new features are live

This PR updates the ‘Send a file by email’ section of the REST-API documentation.

There are 2 stories in the backlog which describe this work:

* [Update 'send a file by email' API client docs for 'email verification' flow](https://www.pivotaltracker.com/story/show/182942256)
* [Update 'send a file by email' API client docs for custom retention periods](https://www.pivotaltracker.com/story/show/182999152)

### CODE REVIEW

File: DOCUMENTATION.md [hunk 1]
```diff
@@ -332,6 +336,82 @@ catch (ApiException $e){}
 catch (InvalidArgumentException $e){}
 ```
 
+#### Ask recipients to confirm their email address before they can download the file
+
+This feature is currently opt-in only. From 29 March 2023 it will apply to all new files by default, unless you opt out.
+
+##### Opt in
```
Comment (karlchillmaid) [inline DOCUMENTATION.md:343]:
```suggestion
##### Opt in

```

File: DOCUMENTATION.md [hunk 2]
```diff
@@ -332,6 +336,83 @@ catch (ApiException $e){}
...
+            'dob'  => '12 July 1968',
+            'link_to_file' => $notifyClient->prepareUpload( $file_data )
+        ]
+    );
+}
+catch (ApiException $e){}
+catch (InvalidArgumentException $e){}
+```
+
+##### Opt out (not recommended)
```
Comment (karlchillmaid) [inline DOCUMENTATION.md:367]:
```suggestion
##### Opt out (not recommended)

```

File: DOCUMENTATION.md [hunk 3]
```diff
@@ -332,6 +336,84 @@ catch (ApiException $e){}
...
+    $file_data = file_get_contents('/path/to/my/file.pdf');
+
+    $response = $notifyClient->sendEmail(
+        'betty@example.com',
+        'df10a23e-2c0d-4ea5-87fb-82e520cbf93c',
+        [
+            'name' => 'Betty Smith',
+            'dob'  => '12 July 1968',
+            'link_to_file' => $notifyClient->prepareUpload( $file_data )
+        ]
```
Comment (karlchillmaid) [inline DOCUMENTATION.md:360]:
```suggestion
            'link_to_file' => $notifyClient->prepareUpload( $file_data ),
            'verify_email_before_download' => 'true'
        ]
```

File: DOCUMENTATION.md [hunk 4]
```diff
@@ -332,6 +336,84 @@ catch (ApiException $e){}
...
```
Comment (karlchillmaid) [inline DOCUMENTATION.md:384]:
```suggestion
            'link_to_file' => $notifyClient->prepareUpload( $file_data ),
            'verify_email_before_download' => 'true'
        ]
```

File: DOCUMENTATION.md [hunk 5]
```diff
@@ -332,6 +336,84 @@ catch (ApiException $e){}
...
```
Comment (karlchillmaid) [inline DOCUMENTATION.md:410]:
```suggestion
            'link_to_file' => $notifyClient->prepareUpload( $file_data ),
            'retention_period' => '4 weeks'
        ]
```

**Comment by samuelhwilliams** [comment]
Merged content into https://github.com/alphagov/notifications-php-client/pull/118

**Pull request closed without merging**